pkg/loadtester/agentloadtester.go

```diff
@@ -0,0 +1,337 @@
+		return "", err
+	}
+	r.echoTrack = echoTrack
+
+	p, err := r.room.LocalParticipant.PublishTrack(echoTrack, &lksdk.TrackPublicationOptions{
+		Name: "echo-track",
+	})
+	if err != nil {
+		return "", err
+	}
+	return p.SID(), nil
+}
+
+func (r *LoadTestRoom) dispatchAgent() error {
+	dispatchClient := lksdk.NewAgentDispatchServiceClient(r.params.URL, r.params.APIKey, r.params.APISecret)
+	req := &livekit.CreateAgentDispatchRequest{
+		Room:      r.room.Name(),
+		AgentName: r.params.AgentName,
+	}
+
+	_, err := dispatchClient.CreateDispatch(context.Background(), req)
+	if err != nil {
+		return err
+	}
+	r.stats.agentDispatchedAt = time.Now()
+	// log.Printf("Successfully dispatched agent %s to room %s", r.params.AgentName, r.room.Name())
+	return nil
+}
+
+func (r *LoadTestRoom) onTrackSubscribed(track *webrtc.TrackRemote, pub *lksdk.RemoteTrackPublication, rp *lksdk.RemoteParticipant) {
+	if r.echoTrack != nil && r.firstParticipant == nil && track.Kind() == webrtc.RTPCodecTypeAudio {
+		log.Printf("Subscribing to echo track in room %s, delay: %s", r.room.Name(), r.params.EchoSpeechDelay.String())
+		r.firstParticipant = rp
+		if rp.Kind() == lksdk.ParticipantAgent {
+			r.stats.agentTrackSubscribed = true
+		}
+
+		type timestampedSample struct {
+			sample   media.Sample
+			received time.Time
+		}
+		sampleChan := make(chan timestampedSample, 1000)
+
+		go func() {
+			for r.running.Load() {
+				pkt, _, err := track.ReadRTP()
+				if err != nil {
+					// log.Printf("Error reading RTP packet: %v", err)
+					continue
+				}
+				sample := timestampedSample{
+					sample: media.Sample{
+						Data:     pkt.Payload,
+						Duration: 20 * time.Millisecond,
+					},
+					received: time.Now(),
+				}
+				sampleChan <- sample
+			}
+			close(sampleChan)
+		}()
+
+		go func() {
+			for r.running.Load() {
+				ts, ok := <-sampleChan
+				if !ok {
+					return
+				}
+				// delay the sample by the echo speech delay
+				delay := time.Until(ts.received.Add(r.params.EchoSpeechDelay))
+				if delay > 0 {
+					time.Sleep(delay)
+				}
+				r.echoTrack.WriteSample(ts.sample, &lksdk.SampleWriteOptions{})
+			}
+		}()
+	}
+}
+
+func (r *LoadTestRoom) onParticipantDisconnected(rp *lksdk.RemoteParticipant) {
+	log.Printf("Participant disconnected, rp:%v/%v", rp.Identity(), rp.SID())
+	if rp.Identity() == r.firstParticipant.Identity() {
+		r.stop()
+	}
+}
+
+func (r *LoadTestRoom) stop() {
+	if !r.isRunning() {
+		return
+	}
+	r.running.Store(false)
+	r.room.Disconnect()
+}
+
+func (t *LoadTestRoom) isRunning() bool {
+	return t.running.Load()
+}
+
+func (t *AgentLoadTester) printStats() {
+	t.lock.Lock()
+	defer t.lock.Unlock()
+
+	checkStyle := lipgloss.NewStyle().Foreground(lipgloss.Color("2")) // Green
+	crossStyle := lipgloss.NewStyle().Foreground(lipgloss.Color("1")) // Red
+
+	table := util.CreateTable().
+		Headers("#", "Room", "Agent Dispatched At", "Agent Joined", "Agent Joined At", "Agent Track Subscribed", "Echo Track Published")
```

Comment:
instead of agent joined at.. would be great to display agent join delay. how long did it take the agent to join?